contracts/GaugeController.sol

```diff
@@ -9,22 +9,68 @@ import "./libraries/ExtendedSafeCastLib.sol";
 contract GaugeController is IGaugeController {
     using ExtendedSafeCastLib for uint256;
 
+    /* ================================================================================ */
+    /* Initialization                                                                   */
+    /* ================================================================================ */
+
     struct GaugeInfo {
         uint256 weight;
     }
 
+    /// @notice ERC20 token contract address (used to weight gauges)
     IERC20 public token;
+
+    /// @notice TokenVault for stakers rewards/incentives
     address public rewardVault;
-    mapping(address => uint256) public rewards;
+
+    /**
+      * @notice Tracks user balance. Balance is used to update target gauge weight balances.
+      * @dev    The balance is updated in deposit, withthdraw, gaugeIncrease and gaugeDecrease.
+    -----------------------------
+    | Address     | Balance     |
+    | ----------- | ----------- |
+    | 0x111...111 | 0x1000      |
+    | 0x222...222 | 0x100000    |
+    -----------------------------
+    */
     mapping(address => uint256) public balances;
+    
+    /**
+      * @notice Tracks User => Gauge => balance.
+      * @dev    The balance is updated in gaugeIncrease and gaugeDecrease.
+    ----------------------------------------------
+    | User        	| Gauge       	| Balance 	|
+    |-------------	|-------------	|---------	|
+    | 0x111...111 	| 0x999...999 	| 0x10000  	|
+    | 0x111...111 	| 0x999...999 	| 0x30000 	|
+    | 0x111...111 	| 0x999...999 	| 0x50000  	|
+    ----------------------------------------------
+    */
     mapping(address => mapping(address => uint256)) public gaugeBalances;
+    
+    /**
+      * @notice Tracks user rewards for staking.
+      * @dev    The rewards is updated in
```

Comment:
Add functions in which the rewards are updated.

Reply:
There are no functions at the moment which update `rewards`